### DISCUSSION TITLE
Tile beacon suddenly disappear

### DISCUSSION THREAD

**Discussion by orzechszek** [theengs/gateway#35] [General]
Hello, 

I have problem with Tile beacon visibility. From time to time notifications in logs disappear. I have to trigger it manually by clicking on it. Can it be somehow triggered using a gateway to keep visibility?

**Comment by 1technophile** [comment]
Hello,

Is there a way maybe in the Tile app to change this behavior ?

**Reply by orzechszek** [reply]
I didn't found such possibility.. sometimes it notifies whole time and sometimes there is no info for a long period.

**Comment by 1technophile** [comment]
When it is working, what messages do you see ?

**Reply by DigiH** [reply]
Not having a Tile myself, so not totally sure, but doesn't a Tile need a connection with its app on a mobile phone to keep sending out pings, so that when it gets out of distance of the phone the app can ring an alert? So possibly once that connection is disabled/the phone app completely closed the Tile then doesn't send regular broadcasts any longer?

Just a thought

**Reply by orzechszek** [reply]
It could be the case... I have it disabled. 
I'm using it to check presence. Do you know any better beacon?

**Reply by DigiH** [reply]
Yes, then it seems to be working nicely within its proprietary functionality. Only transmitting when activated in the app.

Personally I am using simple nrf51822 beacons

https://de.aliexpress.com/item/32884490282.html?spm=a2g0o.detail.1000014.2.5b5125a0CAMqGN&gps-id=pcDetailBottomMoreOtherSeller&scm=1007.40000.267768.0&scm_id=1007.40000.267768.0&scm-url=1007.40000.267768.0&pvid=7cb1e6f9-45d0-4d64-b455-94fe34e4f079&_t=gps-id:pcDetailBottomMoreOtherSeller,scm-url:1007.40000.267768.0,pvid:7cb1e6f9-45d0-4d64-b455-94fe34e4f079,tpp_buckets:668%232846%238116%23946&pdp_ext_f=%7B%22sku_id%22%3A%2265635442588%22%2C%22sceneId%22%3A%2230050%22%7D&pdp_npi=2%40dis%21EUR%21%217.87%21%21%21%21%21%40211b5dec16570470613091442e2182%2165635442588%21rec

not knowing in which country you are I gave an Aliexpress example. Searching for **NRF51822 Beacon** on your favourite online store should do the trick ;)


And while they are double the price now to when I bought them they are still a lot cheaper than a Tile, have a replaceable CR2032 batteries which last 1-2 years, you can name them with up to 8 characters, set their broadcast interval from 50 ms to 10 s, set their major and minor and uuid. They also used to come in versions either broadcasting the txpower @ 1 m , great for precise triangulation with multiple gateways, or alternatively the battery voltage for general presence detection and advance warning of low battery voltage. Although currently I can only see the txpower version on Aliexpress. Also there might be newer versions around now, possibly with BT5 .

Alternatively you could also use any Theengs Decoder compatible beacon like the  Moko, [WSxx](https://decoder.theengs.io/devices/WS02.html) etc. beacons, with them usually also braodcasting additional information like temperature, humidity etc.

All of the above are beacons which **always** transmit their broadcasts, which I think is what you are looking for for your purpose.

**Reply by orzechszek** [reply]
Thanks for info! It doesn't matter what it broadcasts - just somehow check the presence. I'll try to find something with NRF51822 - 1 year is enough.

**Reply by DigiH** [reply]
They will not work with the Tile app though, just for clarification ;)

**Reply by DigiH** [reply]
And if you go for the generic NRF51822 Beacon, here is the manual for how to set its various options

https://manuals.plus/m/9e6548232128ea5a214101b937cf7c4af5ff5b1c48cfcf79d837a7eef4e90cbf_optim.pdf

**Reply by orzechszek** [reply]
> They will not work with the Tile app though, just for clarification ;)

I'm aware :), i just need to have it visible by bt dongle connected with Thenngs Gateway and propagate presence to mosquitto.

**Reply by orzechszek** [reply]
> And while they are double the price now to when I bought them they are still a lot cheaper than a Tile, have a replaceable CR2032 batteries which last 1-2 years, you can name them with up to 8 characters, set their broadcast interval from 50 ms to 10 s, set their major and minor and uuid. 

Did you change broadcast interval? Do you know how to do it?

**Reply by DigiH** [reply]
Yes, I changed mine to every 4 seconds. With the default **Scan_duration** of 10 seconds in OpenMQTTGateway this works nicely to get registered with every scan.

> Do you know how to do it?

In the manual I linked to above, on page 4, section 6 in the table

<img width="698" alt="Screenshot 2022-07-05 at 22 48 08" src="https://user-images.githubusercontent.com/17110652/177413750-856f95ab-c894-44ee-9c5b-fe14dc9e8933.png">

So if hex 02 (also 2 decimal) is 100 ms, then 4 seconds = 4000 ms = 40 * 100 ms => 40 * 2 decimal = 80 decimal = 50 hex

So setting this to **0x16+0x50** with any capable mobile Bluetooth app will set the interval to 4 seconds.

Or simpler, entering **1650** as hex specified to WRITE to the beacon will set it to 4 seconds broadcast interval.

Writing 0x11+name(length<=8) sets the name, e.g.

**114142434445464748** sets the name to **ABCDEFGH**

https://www.calculators.tech/ascii-to-hex

The same with all the other settings in the manual.

**Reply by orzechszek** [reply]
ok, it seems I need ESP to update it. I need to figure it out..

**Reply by DigiH** [reply]
No, you don't need an ESP.

Any free Bluetooth smart phone app which can connect and READ/WRITE to the beacon can do this. nRF Connect, BT Inspector, LightBlue …

But if you possibly find it too difficult any other of the Decoder/Gateway compatible beacon sensors also transmitting temperature will do the trick, although I don't know exactly what their transmission intervals are.

**Reply by orzechszek** [reply]
It seems nRF Connect and LightBlue  doesn't recognize all the parameters of the beacon :(  . I can't find interval or possibility to send hex to sensor. Are you able to check how it looks with your module?

**Reply by orzechszek** [reply]
ok, I found something - if I proper understand I should send it there:

![image](https://user-images.githubusercontent.com/34706669/180966611-3c3f435c-6262-4437-a279-3907d56df9bd.png)

with LightBlue  I was able to send it with hex - **1650**, but nothing has changed.. it still broadcasts with 1s.

![IMG_20220726_105312](https://user-images.githubusercontent.com/34706669/180968105-9bb8e737-aa6c-41a1-88b3-f374c391d5e9.jpg)

**Reply by DigiH** [reply]
Yes, that seems the correct page you found. BTW, I also saw that they updated their iOS app under a different name, as my old version didn't want to work any longer. If you're on iOS you could get

https://apps.apple.com/us/app/rl-beacon/id1191040734

or see if they have released a similar version for Android.

Alternatively on your screen above, type in
`
1154455354`

and push the **WRITE** button. This should set the beacon name to **TEST**

11 for setting the name, the next four hex bytes for TEST. The same with a leading 16 and the appropriate number in hex for the interval, and so on as stated in the manual I linked to above.

**Reply by orzechszek** [reply]
Unfortunately I'm using android. I can't find an app for my phone..

**Reply by DigiH** [reply]
You might not find the RLBeacon app for Android, but the screenshot you sent above from the other app looks good. 

So did you try writing in the field where you had entered **1650**, and pushed on WRITE?

**Reply by DigiH** [reply]
How does the beacon show up in your MQTT broker through Theengs Gateway now?

**Reply by DigiH** [reply]
Here is another app i can use for writing the settings on iOS and it has an Android version, but really any capable BLE app will do it, as the one in your screenshot as well.

https://play.google.com/store/apps/details?id=com.macdom.ble.blescanner

**Reply by DigiH** [reply]
@orzechszek  How do you know it still broadcasts with 1s after you sent the 1650 code?

**Reply by orzechszek** [reply]
> You might not find the RLBeacon app for Android, but the screenshot you sent above from the other app looks good.
> 
> So did you try writing in the field where you had entered **1650**, and pushed on WRITE?

Yes, as I show you in screenshot - enter 1650 and push WRITE, still the same broadcast time.

> Here is another app i can use for writing the settings on iOS and it has an Android version, but really any capable BLE app will do it, as the one in your screenshot as well.
> 
> https://play.google.com/store/apps/details?id=com.macdom.ble.blescanner

I've installed it, but in android version there is no hex, just byte



> @orzechszek How do you know it still broadcasts with 1s after you sent the 1650 code?

In nRF Connector when you click SCAN, you can notice information in miliseconds under CONNECT button - it seems it's boadcast time. Also, when you click on beacon and next on MORE, then HISTORY table appears with historical broadcast data.

**Reply by DigiH** [reply]
> I've installed it, but in android version there is no hex, just byte

Could you post a screenshot of this when you are in the **Write,WriteWithoutResponse**, and also of the **Advertisment Data** of the beacon in the main window of BLE Scanner after you connected to it?

**Reply by orzechszek** [reply]
> > I've installed it, but in android version there is no hex, just byte
> 
> Could you post a screenshot of this when you are in the **Write,WriteWithoutResponse**, and also of the **Advertisment Data** of the beacon in the main window of BLE Scanner after you connected to it?

![IMG_20220727_134113](https://user-images.githubusercontent.com/34706669/181253280-5eaddea4-924a-4fd9-b341-04d25020e807.jpg)

**Reply by DigiH** [reply]
Ok, slightly different to my RDL51822 with an additional service 8E40000…, but also the Advertisment Data should be further up, probably also with an option to open it up, or also open up **Generic Attribute**. Would be good to also see what is stated there.

But for writing it should be the very bottom characteristic in the the screenshot. Can you also make a screenshot when going into it (00001805-…)?

**Reply by orzechszek** [reply]
**Generic Attribute** is empty.

In 00001805-… I just can click **W** icon and provide **BYTE ARRAY** data.

**Reply by DigiH** [reply]
So what happens when you provide

1154455354

and confirm? The beacon should then have the name **TEST** instead of **RDL51822**. Possibly needs to turn if OFF and ON again with the button on the beacon.

**Reply by DigiH** [reply]
Or is it on Android that you might have to enter it with the notation
`0x110x540x450x530x54`
???

**Reply by DigiH** [reply]
Just to show you what I mean, on iOS in the very same BLE Scanner app I get the following when clicking on the WRITE option, being able to chose between **Hex** or **Text** entry

![IMG_1519](https://user-images.githubusercontent.com/17110652/181273201-6f7a794c-9591-4f67-939e-96663a8d89f0.PNG)

**Reply by orzechszek** [reply]
In android app there is **BYTE ARRAY** and unfortunately it's not working, name stays the same.

![IMG_20220727_164400](https://user-images.githubusercontent.com/34706669/181277360-6189842b-437d-4cd5-afb3-c19c3236c0c7.jpg)

**Reply by DigiH** [reply]
Yes, and when you chose **BYTE ARRAY** and then enter the above suggest hex bytes

`1154455354`

it doesn't change the name,? Or even with the, I don't really think necessary, notation with

`0x110x540x450x530x54`

**Reply by DigiH** [reply]
Also in your screenshot in the background I can see that the **Status: DISCONNECTED**.

The writing only will work when connected, so don't wait too long and make sure it is **Status: CONNECTED**, like in my screenshot above.

Just go out to the main screen and connect again.

**Reply by orzechszek** [reply]
Yes, during pushing data it's connected, just this screenshot took some time.

**Reply by DigiH** [reply]
Then I'm afraid I really haven't got any idea what might be causing the issue on Android for you :(

@koenvervloesem, do you have an Android smart phone, and does setting the parameters work for your RadioLand beacon?

**Reply by koenvervloesem** [reply]
My RDL52832 seems to have the same characteristics, but writing the byte array 1154455354 with nRF Connect for Mobile on Android to characteristic 1805 doesn't change the beacon's advertised name... The device also disconnects immediately, often even before I'm able to write the byte array, and  otherwise immediately after writing. So no luck here either :-(

**Reply by DigiH** [reply]
Thanks for the confirmation. Hmm :( this is all very strange, as all my 6 RDL51822 beacons works nicely this way, to change their name, major, minor, interval ... and the RDL51822 manual, as well as the RDL52832 both describe this as being the correct way to set all the parameters.

My RDL51822 have been ordered a long time ago, so unless RadioLand has made some changes, or there are unauthorised/not correctly implemented _fake_ versions being sold on Aliexpress I'll have to wait and see when my ordered RDL52832 arrives, to see how it behaves.

**Reply by orzechszek** [reply]
> My RDL52832 seems to have the same characteristics, but writing the byte array 1154455354 with nRF Connect for Mobile on Android to characteristic 1805 doesn't change the beacon's advertised name... The device also disconnects immediately, often even before I'm able to write the byte array, and otherwise immediately after writing. So no luck here either :-(

Exactly the same behavior like yours.. :(.

**Reply by DigiH** [reply]
Hi @orzechszek @koenvervloesem I've just received my RDL52832 today, and am experiencing the same issue as you, no tobeing able set of any parameters.

But having seen the beacon in the nrfConnect app I immediately realised the difference to my older RDL51822, which I ordered several years ago. So my new RDL52832 was delivered in DFU mode, for being able to update the firmware straight after delivery. This obviously prevents from any proper BT connection with being able to set any parameter :( seems that they are all sent out like that these days.

Can you confirm that you are seeing the same with your RDL52832 and RDL51822?

![DFU](https://user-images.githubusercontent.com/17110652/181813994-2cde7dcc-525b-43db-a0cc-addb0ba2c235.png)

Now all we need to find out now is how to be able to get out of the DFU mode, back to the application mode easily. Only quick link I found so far

https://devzone.nordicsemi.com/f/nordic-q-a/10884/how-to-exit-dfu-mode

https://devzone.nordicsemi.com/f/nordic-q-a/81926/nrf52832-won-t-reset-chip-after-dfu

https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_nrf5_v17.0.2%2Flib_dfu_transport.html&cp=8_5_0_3_5_2

**Reply by orzechszek** [reply]
Yes, mine is in DFU mode.
It seems there is a need to send a reset command - do you know how to do it with android app?

**Reply by DigiH** [reply]
Unfortunately not, no experience with flashing Nordic Semiconductor beacons so far, but am reading through some of their GitHub pages, and also nrfConnect allows for OTA DFU firmware updates. So maybe if we find a suitable package they might restart into application after re-flashing.

All hypothetical so far, needs more reading up ;) but we should be able to eventually come out of the DFU mode and everything should work as expected with being able to set parameters in the app mode.

**Reply by koenvervloesem** [reply]
Yes, it's the same situation here.

**Reply by DigiH** [reply]
So that seems to be the norm when receiving RDL beacons from China these days :unamused:

I still haven't got any further with a solution yet :( only that the nRF Toolbox app correctly recognises the DFU mode and would allow a package install, if we only knew how and where to get correct ones for our models, without having to dive fully into beacon development ;)

@orzechszek so apart from currently not being able to set the name and interval, you can still use the beacon for your intended use, right?

**Reply by orzechszek** [reply]
Yes, except faster battery consuming it's working as expected 😀

**Comment by koenvervloesem** [comment]
I don't have a Tile, but isn't this a Proximity Reporter? If so, it only advertises when it isn't connected to the app on your phone. What does nRF Connect show you for the device? Or what is the MQTT output of Theengs Gateway or the output of Theengs Explorer?
